Mathlib/Analysis/NormedSpace/WeakOperatorTopology.lean

```diff
@@ -0,0 +1,283 @@
+of this topology. In particular, we show that it is a topological vector space.
+-/
+section Topology
+
+variable (𝕜) (E) (F) in
+/-- The function that induces the topology on `E →WOT[𝕜] F`, namely the function that takes
+an `A` and maps it to `fun ⟨x, y⟩ => y (A x)` in `E × F⋆ → 𝕜`, bundled as a linear map to make
+it easier to prove that it is a TVS. -/
+def inducingFn : (E →WOT[𝕜] F) →ₗ[𝕜] (E × F⋆ → 𝕜) where
+  toFun := fun A ⟨x, y⟩ => y (A x)
+  map_add' := fun x y => by ext; simp
+  map_smul' := fun x y => by ext; simp
+
+/-- The weak operator topology is the coarsest topology such that `fun A => y (A x)` is
+continuous for all `x, y`. -/
+instance instTopologicalSpace : TopologicalSpace (E →WOT[𝕜] F) :=
+  .induced (inducingFn _ _ _) Pi.topologicalSpace
+
+@[fun_prop]
+lemma continuous_inducingFn : Continuous (inducingFn 𝕜 E F) :=
+  continuous_induced_dom
+
+lemma continuous_dual_apply (x : E) (y : F⋆) : Continuous fun (A : E →WOT[𝕜] F) => y (A x) := by
+  refine (continuous_pi_iff.mp continuous_inducingFn) ⟨x, y⟩
+
+@[fun_prop]
+lemma continuous_of_dual_apply_continuous {α : Type*} [TopologicalSpace α] {g : α → E →WOT[𝕜] F}
+    (h : ∀ x (y : F⋆), Continuous fun a => y (g a x)) : Continuous g :=
+  continuous_induced_rng.2 (continuous_pi_iff.mpr fun p => h p.1 p.2)
+
+lemma embedding_inducingFn : Embedding (inducingFn 𝕜 E F) := by
+  refine Function.Injective.embedding_induced fun A B hAB => ?_
+  rw [ext_dual_iff]
+  simpa [Function.funext_iff] using hAB
+
+open Filter in
+/-- The defining property of the weak operator topology: a function `f` tends to
+`A : E →WOT[𝕜] F` along filter `l` iff `y (f a x)` tends to `y (A x)` along the same filter. -/
+lemma tendsto_iff_forall_dual_apply_tendsto {α : Type*} {l : Filter α} {f : α → E →WOT[𝕜] F}
+    {A : E →WOT[𝕜] F} :
+    Tendsto f l (𝓝 A) ↔ ∀ x (y : F⋆), Tendsto (fun a => y (f a x)) l (𝓝 (y (A x))) := by
+  have hmain : (∀ x (y : F⋆), Tendsto (fun a => y (f a x)) l (𝓝 (y (A x))))
+      ↔ ∀ (p : E × F⋆), Tendsto (fun a => p.2 (f a p.1)) l (𝓝 (p.2 (A p.1))) :=
+    ⟨fun h p => h p.1 p.2, fun h x y => h ⟨x, y⟩⟩
+  rw [hmain, ← tendsto_pi_nhds, Embedding.tendsto_nhds_iff embedding_inducingFn]
+  rfl
+
+lemma le_nhds_iff_forall_dual_apply_le_nhds {l : Filter (E →WOT[𝕜] F)} {A : E →WOT[𝕜] F} :
+    l ≤ 𝓝 A ↔ ∀ x (y : F⋆), l.map (fun T => y (T x)) ≤ 𝓝 (y (A x)) :=
+  tendsto_iff_forall_dual_apply_tendsto (f := id)
+
+instance instT3Space : T3Space (E →WOT[𝕜] F) := embedding_inducingFn.t3Space
+
+instance instContinuousAdd : ContinuousAdd (E →WOT[𝕜] F) := .induced (inducingFn 𝕜 E F)
+instance instContinuousNeg : ContinuousNeg (E →WOT[𝕜] F) := .induced (inducingFn 𝕜 E F)
+instance instContinuousSMul : ContinuousSMul 𝕜 (E →WOT[𝕜] F) := .induced (inducingFn 𝕜 E F)
+
+instance instTopologicalAddGroup : TopologicalAddGroup (E →WOT[𝕜] F) where
+
+instance instUniformSpace : UniformSpace (E →WOT[𝕜] F) := .comap (inducingFn 𝕜 E F) inferInstance
```

Comment:
What about completeness? This should hold at least when `E` and `F` are Hilbert spaces.

Reply:
Yeah, I think I'll leave that for a future PR!